include/boost/url/detail/impl/remove_dot_segments.ipp

```diff
@@ -0,0 +1,518 @@
+        }
+        else if (l)
+        {
+            if (c.empty())
+            {
+                c = "/..";
+            }
+            else
+            {
+                c = "/../";
+            }
+            --l;
+            return;
+        }
+        c = {};
+    };
+
+    // Calculate the normalized size
+    auto norm_bytes = [&pop_last](
+         string_view p,
+         bool r)
+    {
+        string_view c;
+        std::size_t s{0};
+        std::size_t n{0};
+        do
+        {
+            pop_last(p, c, s, r);
+            n += pct_decode_bytes_unchecked(c);
+        } while (!c.empty());
+        return n;
+    };
+    std::size_t s0n = norm_bytes(
+        s0,
+        s0_remove_unmatched);
+    if (!s0_remove_unmatched)
+        s0n += s0_prefix_n;
+
+    std::size_t rhs_n = norm_bytes(
+        s1,
+        s1_remove_unmatched);
+    if (!s1_remove_unmatched)
+        rhs_n += s1_prefix_n;
+
+    // Remove child segments until last intersection
+    s0 = s0_init;
+    s1 = s1_init;
+    string_view s0c;
+    string_view s1c;
+    std::size_t s0l = 0;
+    std::size_t s1l = 0;
+    std::size_t s0i = s0n;
+    std::size_t s1i = rhs_n;
+    pop_last(
+        s0, s0c, s0l,
+        s0_remove_unmatched);
+    pop_last(
+        s1, s1c, s1l,
+        s1_remove_unmatched);
+
+    // Consume incomparable segments
+    auto pop_decoded_back = [](string_view& s)
+    {
+        if (s.size() < 3 ||
+            *std::prev(s.end(), 3) != '%')
+        {
+            char c = s.back();
+            s.remove_suffix(1);
+            return c;
+        }
+        char c = 0;
+        pct_decode_unchecked(
+            &c, &c + 1, s.substr(s.size() - 3));
+        s.remove_suffix(3);
+        return c;
+    };
+
+    while (s0i != s1i)
+    {
+        // Consume more child segments
+        if (s0c.empty())
+        {
+            pop_last(
+                s0, s0c, s0l, s0_remove_unmatched);
+        }
+        if (s1c.empty())
+        {
+            pop_last(
+                s1, s1c, s1l, s1_remove_unmatched);
+        }
+
+        // Remove incomparable suffix
+        while (!s0c.empty() &&
+               !s1c.empty())
+        {
+            if (s1i > s0i)
+            {
+                pop_decoded_back(s1c);
+                --s1i;
+                continue;
+            }
+            else if (s0i > s1i)
+            {
+                pop_decoded_back(s0c);
+                --s0i;
+                continue;
+            }
+            break;
+        }
+    }
+
+    int cmp = 0;
+    BOOST_ASSERT(s0i == s1i);
+    while (s0i > 0)
+    {
+        // Consume more child segments
+        if (s0c.empty())
+        {
+            pop_last(
+                s0, s0c, s0l, s0_remove_unmatched);
+        }
+        if (s1c.empty())
+        {
+            pop_last(
+                s1, s1c, s1l, s1_remove_unmatched);
+        }
+
+        // Compare intersection
+        while (!s0c.empty() &&
+               !s1c.empty())
```

Comment:
weird format

Reply:
Why?

Reply:
the attempt at vertical alignment
```
        while(
            !s0c.empty() &&
            !s1c.empty());
```

I usually never vertically align, because it puts things off the tab stops (e.g. line 467 is indented 15 spaces)